data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/ConsumerVerticleBuilder.java

```diff
@@ -126,11 +117,9 @@ private void build(final Vertx vertx, final ConsumerVerticle consumerVerticle, f
     );
     consumerVerticle.setRecordDispatcher(recordDispatcher);
 
-    final var partitionRevokedHandlers = List.of(
-      consumerVerticle.getPartitionRevokedHandler(),
-      offsetManager.getPartitionRevokedHandler()
-    );
-    this.consumerVerticleContext.withConsumerRebalanceListener(createRebalanceListener(partitionRevokedHandlers));
+    consumerVerticle.addPartitionRevokedHandler(consumerVerticle.getPartitionRevokedHandler());
+    consumerVerticle.addPartitionRevokedHandler(offsetManager.getPartitionRevokedHandler());
```

Comment:
Did you consider passing to the consumer verticle the rebalance listener instead of passing all revoke handlers?

That would decouple the consumer verticle from from the rebalance logic